src/components/FiltersDrawer/Filter/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+import Button, { Variant } from "@leafygreen-ui/button";
+import IconButton from "@leafygreen-ui/icon-button";
+import { palette } from "@leafygreen-ui/palette";
+import {
+  SegmentedControlOption as Option,
+  SegmentedControl,
+} from "@leafygreen-ui/segmented-control";
+import TextInput from "@leafygreen-ui/text-input";
+import { Body } from "@leafygreen-ui/typography";
+import Icon from "components/Icon";
+import { size } from "constants/tokens";
+
+const { gray } = palette;
+
+enum CaseSensitivity {
+  Sensitive = "sensitive",
+  Insensitive = "insensitive",
+}
+
+enum MatchType {
+  Exact = "exact",
+  Inverse = "inverse",
+}
+
+interface FilterProps {
+  ["data-cy"]?: string;
+  filterName: string;
+  deleteFilter: (filter: string) => void;
+  editFilter: (oldFilter: string, newFilter: string) => void;
+}
+
+const Filter: React.FC<FilterProps> = ({
+  "data-cy": dataCy,
+  filterName,
+  deleteFilter,
+  editFilter,
+}) => {
+  const [newFilter, setNewFilter] = useState("");
+  const [isEditing, setIsEditing] = useState(false);
+  const [isVisible, setIsVisible] = useState(true);
+  const [caseSensitivity, setCaseSensitivity] = useState<string>(
+    CaseSensitivity.Insensitive
+  );
+  const [matchType, setMatchType] = useState<string>(MatchType.Exact);
+
+  const resetEditState = () => {
+    setIsEditing(false);
+    setNewFilter("");
+  };
+
+  return (
+    <FilterRow data-cy={dataCy}>
+      <FilterLabel>
+        <Badge>FILTER</Badge>
+        <FilterIcons>
+          <IconButton
+            aria-label="Edit filter button"
```

Comment:
[nit] Aria label text doesn't need to announce the role of the element, it can just describe its function briefly. There are some good guidelines [here](https://www.aditus.io/aria/aria-label/#when-to-use)!
```suggestion
            aria-label="Edit filter"
```